### PR TITLE
Fix NRE in VisualBasicAddExplicitCastCodeFixProvider 

### PR DESCRIPTION
**Pull request by kfechter**
Fixes issue #61538

NRE no longer occurs when a class attribute is used that triggers BC30519. Cast suggestions are now provided. 

image

### CODE REVIEW

File: src/Analyzers/VisualBasic/CodeFixes/AddExplicitCast/VisualBasicAddExplicitCastCodeFixProvider.vb
Comment (kfechter) [reply src/Analyzers/VisualBasic/CodeFixes/AddExplicitCast/VisualBasicAddExplicitCastCodeFixProvider.vb:78]: Unclear as to what this change is for. 

Not against making the change, Just need clarification on why this is better than the code that was originally there.

Comment (CyrusNajmabadi) [reply src/Analyzers/VisualBasic/CodeFixes/AddExplicitCast/VisualBasicAddExplicitCastCodeFixProvider.vb:78]: instead of checking every arbitrary node, you are only looking at invocations.  and because you're only looking at invocatinos, you don't have to enumerate all child nodes looking for argumentlists, you can just directly check if it has an argument list or not.

(the same feedback applies to the check below this for atttributes as well).

thanks!

Comment (kfechter) [reply src/Analyzers/VisualBasic/CodeFixes/AddExplicitCast/VisualBasicAddExplicitCastCodeFixProvider.vb:78]: Gotcha! I'll make that change tonight and test it once I'm off work

thanks.

**Comment by genlu** [comment]
Hey @kfechter, thanks for your contribution! Could you please also add a unittest [here](https://github.com/dotnet/roslyn/blob/main/src/Analyzers/VisualBasic/Tests/AddExplicitCast/AddExplicitCastTests.vb) to covers the scenario?

**Comment by kfechter** [comment]
> Hey @kfechter, thanks for your contribution! Could you please also add a unittest [here](https://github.com/dotnet/roslyn/blob/main/src/Analyzers/VisualBasic/Tests/AddExplicitCast/AddExplicitCastTests.vb) to cover the scenario?

I'm trying to work through adding this unit test but I'm hitting a bit of a roadblock. I'm positive this is just due to my not understanding how the unit tests work, but I've added one by looking at what the others are doing and trying to replicate that with this scenario, but the test just doesn't seem to be working. 

The below is what I am trying to do to implement this test. 

```
        <Fact(), Trait(Traits.Feature, Traits.Features.CodeActionsAddExplicitCast)>
        Public Async Function TestClassAttribute() As Task
            Await TestInRegularAndScriptAsync(
"Option Strict On
Class Program
    <System.Runtime.InteropServices.ClassInterface(|1|)>
    Class Base
    End Class
End Class",
"Option Strict On
Class Program
    <System.Runtime.InteropServices.ClassInterface(CShort(1))>
    Class Base
    End Class
End Class")
        End Function
```

however, with the above code, I get the following error
    System.InvalidOperationException : Sequence contains no elements

This is my first time delving into these unit tests, so I'm positive I'm just missing something super simple

**Comment by dibarbet** [comment]
> however, with the above code, I get the following error
> System.InvalidOperationException : Sequence contains no elements

This might not be the only cause, but I think the carrot location marker is missing brackets.  Your code shows 
```
ClassInterface(|1|)
```
but I believe it should be
```
ClassInterface([|1|])
```
e.g. https://github.com/dotnet/roslyn/blob/main/src/Analyzers/VisualBasic/Tests/AddExplicitCast/AddExplicitCastTests.vb#L30

**Comment by kfechter** [comment]
> > however, with the above code, I get the following error
> > System.InvalidOperationException : Sequence contains no elements
> 
> This might not be the only cause, but I think the caret location marker is missing brackets. Your code shows
> 
> ```
> ClassInterface(|1|)
> ```
> 
> but I believe it should be
> 
> ```
> ClassInterface([|1|])
> ```
> 
> e.g. https://github.com/dotnet/roslyn/blob/main/src/Analyzers/VisualBasic/Tests/AddExplicitCast/AddExplicitCastTests.vb#L30

ah ha. I'm getting further now. It throws "No action was offered when one was expected." but i'm probably missing something in the class definition that the test framework is expecting.

File: src/Analyzers/VisualBasic/Tests/AddExplicitCast/AddExplicitCastTests.vb
Comment (kfechter) [reply src/Analyzers/VisualBasic/Tests/AddExplicitCast/AddExplicitCastTests.vb:487]: ah, that explains why I couldn't get the test to fail. I will revert that in the morning. I left the power adapter for my Dev laptop at work.

File: src/Analyzers/VisualBasic/CodeFixes/AddExplicitCast/VisualBasicAddExplicitCastCodeFixProvider.vb
Comment (CyrusNajmabadi) [inline src/Analyzers/VisualBasic/CodeFixes/AddExplicitCast/VisualBasicAddExplicitCastCodeFixProvider.vb:80]:
```suggestion
                    Dim attribute= spanNode.GetAncestors(Of AttributeSyntax).FirstOrDefault(
```

File: src/Analyzers/VisualBasic/Tests/AddExplicitCast/AddExplicitCastTests.vb
Comment (kfechter) [reply src/Analyzers/VisualBasic/Tests/AddExplicitCast/AddExplicitCastTests.vb:487]: Reverted and set Option Strict Off, but now I'm getting

 System.Exception : No action was offered when one was expected.

I debugged, and it looks like the CodeAction is null, but i'm not understanding how the test framework actually gets the CodeActions to perform. below is my most recent test code. 

```
        <Fact(), Trait(Traits.Feature, Traits.Features.CodeActionsAddExplicitCast)>
        Public Async Function TestClassAttribute() As Task
            Await TestInRegularAndScriptAsync(
"Option Strict Off
Imports System.Runtime.InteropServices
<ClassInterface([|1|])>
Class Program
End Class",
"Option Strict Off
Imports System.Runtime.InteropServices
<ClassInterface(CShort(1))>
Class Program
End Class")
        End Function
```

Comment (CyrusNajmabadi) [reply src/Analyzers/VisualBasic/Tests/AddExplicitCast/AddExplicitCastTests.vb:487]: i'd recommend not testing this exact interface, but rather just include a simple copy of it in the test.

Comment (dibarbet) [reply src/Analyzers/VisualBasic/Tests/AddExplicitCast/AddExplicitCastTests.vb:487]: @kfechter I finally got it working - looks like I needed to pass the option strict as a compilation option instead.  The change I made is here - https://github.com/dibarbet/roslyn/commit/5cf383cefddb499138d0fb154f06104921e26f35

I can either update your branch here or I can leave it to you to update the test, whichever you prefer.

Comment (kfechter) [reply src/Analyzers/VisualBasic/Tests/AddExplicitCast/AddExplicitCastTests.vb:487]: I can update the test. I appreciate the help!

**Review by dibarbet — APPROVED** [review]
lgtm!  Our main branch is locked down due to some infra issues, but I'll merge this when it opens.  Thanks!!

**Pull request merged**
